app/models/settings/authentication.rb

```diff
@@ -0,0 +1,44 @@
+module Settings
+  class Authentication < RailsSettings::Base
+    self.table_name = :settings_authentications
+
+    # The configuration is cached, change this if you want to force update
+    # the cache, or call Settings::Authentication.clear_cache
+    cache_prefix { "v1" }
+
+    field :allow_email_password_login, type: :boolean, default: true
+    field :allow_email_password_registration, type: :boolean, default: false
+    field :allowed_registration_email_domains, type: :array, default: %w[], validates: {
+      valid_domain_csv: true
+    }
+    field :apple_client_id, type: :string
+    field :apple_key_id, type: :string
+    field :apple_pem, type: :string
+    field :apple_team_id, type: :string
+    field :display_email_domain_allow_list_publicly, type: :boolean, default: false
+    field :facebook_key, type: :string
+    field :facebook_secret, type: :string
+    field :github_key, type: :string, default: ApplicationConfig["GITHUB_KEY"]
+    field :github_secret, type: :string, default: ApplicationConfig["GITHUB_SECRET"]
+    field :invite_only_mode, type: :boolean, default: false
+    field :providers, type: :array, default: %w[]
+    field :require_captcha_for_email_password_registration, type: :boolean, default: false
+    field :twitter_key, type: :string, default: ApplicationConfig["TWITTER_KEY"]
+    field :twitter_secret, type: :string, default: ApplicationConfig["TWITTER_SECRET"]
+
+    # Google ReCATPCHA keys
+    field :recaptcha_site_key, type: :string, default: ApplicationConfig["RECAPTCHA_SITE"]
+    field :recaptcha_secret_key, type: :string, default: ApplicationConfig["RECAPTCHA_SECRET"]
```

Comment:
should these be part of authentication? I'm okay with leaving them here but they seem more related to "authorization" than "authentication". Not advocating for a separate model, just a thought

Reply:
Yeah, I was debating that with myself. In the end it felt "close enough" and I wanted to slim down `SiteConfig` a bit more. You do have a good point though and I'm not opposed to moving them back if others think it's better.

Reply:
Ugh, I now spent some time changing this back and then remembered why I moved them, they live within the same form which now submits to a different controller:

![image](https://user-images.githubusercontent.com/47985/113092657-a0d33a00-9218-11eb-80c2-db0d5412fd3d.png)